tools/terraform/provider/provider.go

```diff
@@ -0,0 +1,146 @@
+package provider
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"os"
+	"terraform-provider-fleetdm/fleetdm_client"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// This Terraform provider is based on the example provider from the Terraform
+// documentation. It is a simple provider that interacts with the FleetDM API.
+
+// Ensure FleetDMProvider satisfies various provider interfaces.
+var _ provider.Provider = &FleetDMProvider{}
+var _ provider.ProviderWithFunctions = &FleetDMProvider{}
```

Comment:
Not familiar with implementing a terraform provider, I'll just skim those files quickly.